spec.bs

```diff
@@ -5369,15 +5429,77 @@ result of [=evaluating a bidding script=], or an [=additional bid=] provided by
   : <dfn>component seller</dfn>
   :: Null or an [=origin=]. Seller in component auction which the [=generated bid/interest group=]
     is participating. Only set for component auctions, null otherwise.
+  : <dfn>target number of ad components</dfn>
+  :: Null or {{unsigned long}}, initially null. Set if the bidder requested the component list
+    they provide to be reduced to the given target, taking in account k-anonymity.
+  : <dfn>number of mandatory ad components</dfn>
+  :: {{unsigned long}}. Only relevant if [=generated bid/target number of ad components=] is set.
+     Requires the first that many given ad components to be included.
 </dl>
 
+<div algorithm>
+
+To <dfn>apply any component ads target to a bid</dfn>, given a [=generated bid=] |generatedBid|:
+1. If |generatedBid|'s [=generated bid/target number of ad components=] is null, return.
+1. [=Assert=] that |generatedBid|'s [=generated bid/ad component descriptors=] is not null.
+1. Truncate |generatedBid|'s [=generated bid/ad component descriptors=] to length [=generated bid/target number of ad components=].

```

Comment:
@domfarolino Is there a standard way of truncating a list to a certain length?

Reply:
Not really. I would just do a while loop where you repeatedly `[=list/remove=]` the last value from the list until the `[=list/size=]` is the number you want it to be.

Reply:
Done.